x/bank/keeper/keeper.go

```diff
@@ -287,6 +288,29 @@ func (k BaseKeeper) SendCoinsFromAccountToModule(
 	return k.SendCoins(ctx, senderAddr, recipientAcc.GetAddress(), amt)
 }
 
+// SendCoinsFromModuleToAccountOriginalVesting does existing bank keeper logic for SendCoinsFromModuleToAccount,
+// except the recipient account should have its original vesting increment if its a vesting account.
+// If recipient is a regular account, just do a normal send
+func (k BaseKeeper) SendCoinsFromModuleToAccountOriginalVesting(
+	ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins,
+) error {
+
+	err := k.SendCoinsFromModuleToAccount(ctx, senderModule, recipientAddr, amt)
```

Comment:
We need to check if its a normal account before doing this, right?

Reply:
Vesting account is a kind of normal account isn't it?

Reply:
We confirmed over call, we think this is right. It computes things as bank balance - locked balance, so we have to increment both.

Reply:
We want to add a test for just ensuring these assumptions - nothing becomes immediately spendable, and the full amount is spendable at the end time.

Reply:
Hmm not failing here.
https://github.com/osmosis-labs/cosmos-sdk/pull/4/files#diff-04e4d8ff2ba58020d7cf7f0d80a7505a13719406b6be2ea09ddc402ff412a0e6R829

Reply:
I don't think I understand this test correctly